lib/std/crypto.zig

```diff
@@ -109,8 +109,16 @@ pub const onetimeauth = struct {
 ///
 /// Password hashing functions must be used whenever sensitive data has to be directly derived from a password.
 pub const pwhash = struct {
+    pub const Encoding = enum {
+        phc,
+        crypt,
+    };
+
     pub const bcrypt = @import("crypto/bcrypt.zig");
     pub const pbkdf2 = @import("crypto/pbkdf2.zig").pbkdf2;
```

Comment:
Should this import be consistent with bcrypt and scrypt ones?

Reply:
It doesn't have any public methods like `strHash` or `strVerify` and I don't know if they are needed. For now If we make it the same as `bcrypt` and `scrypt` it will have only one public function named exactly the same as variable and calling it will be like this:

```zig
const std = @import("std");
const pbkdf2 = std.crypto.pwhash.pbkdf2;

const HmacSha256 = crypto.auth.hmac.sha2.HmacSha256;

fn main() !void {
    var dk: [32]u8 = undefined;
    try pbkdf2.pbkdf2(dk, "password", "salt", 1, HmacSha256);
}
```

Of cause we can do the same as with `bcrypt` and `scrypt` and wrap `rounds` and `Prf` params with struct. Then it will have `Params` public struct and `pbkdf2` function. Also we may call all `kdf` or `hash` functions the same across `pwhash`. Maybe @jedisct1 has something to say.

Reply:
Right. PBKDF2's output gets serialized, but in many different application-specific ways. Probably not something suitable for a standard library. It's also a generic construction and not a function.
So, we should probably just keep it the way it is.